### PR TITLE
Wip libraryitem

### PR DESCRIPTION
**Pull request by cvrkota**
Adding ability to add resource to the database, to view one resource at the time, and to view all resources from the library

### CODE REVIEW

File: db/migrate/20200917005322_create_items.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class CreateItems < ActiveRecord::Migration[6.0]
+  def change
+    create_table :items do |t|
+      t.string :new
```
Comment (spinecone) [inline db/migrate/20200917005322_create_items.rb:4]: Are there any other fields we might want on this table? Thinking about things like the type of media (is it a youtube video, library book, online article, etc.), the url if it's an online resource, the date it was made?

File: db/schema.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# be faster and is potentially less error prone than running all of your
+# migrations from scratch. Old migrations may fail to apply correctly if those
+# migrations use external dependencies or application code.
+#
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema.define(version: 2020_09_17_005322) do
+
+  create_table "items", force: :cascade do |t|
+    t.string "new"
```
Comment (spinecone) [inline db/schema.rb:16]: What is the "new" field going to be used for?

File: app/models/item.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Item < ApplicationRecord
+    validates :title, presence: true
```
Comment (spinecone) [inline app/models/item.rb:2]: Validations for maximum length on these fields could also be useful.

File: app/controllers/items_controller.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  def new
+    @item = Item.new
+  end
+
+  def create
+    @item = Item.new(item_params)
+    if @item.save
+      flash[:success] = "You have added a new resource item to the library!"
+      redirect_to '/items/new'
```
Comment (spinecone) [inline app/controllers/items_controller.rb:19]: Typically after creating something, we redirect to the "show" page for that object. If we'd rather go back to the "new" form instead that's fine too, just wanted to mention it as something that might feel more intuitive for the user.

File: app/controllers/items_controller.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    else
+      render 'new'
+    end
+  end
+
+  private
+
+    def item_params
```
Comment (spinecone) [inline app/controllers/items_controller.rb:27]: Indenting got a little weird here!

File: app/helpers/items_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module ItemsHelper
```
Comment (spinecone) [inline app/helpers/items_helper.rb:1]: I would recommend deleting this file if we don't currently have a use for it.

**Comment by spinecone** [comment]
Let's merge this in after:
- Address removing the "new" field, fixing indenting, and deleting the ItemsHelper
- Wait for the admin login PR to be merged
- Deal with merge conflicts here that will come from that 😭 
- We get a 👍 from one of the reviewers

File: app/views/items/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<% provide(:title, 'All resources') %> 
+<h1>All library resources</h1>
+    <ul class="items">
+        <% @items.each do |item| %>
+            <li>
+            <%= link_to item.title, item %>
+            </li> 
+        <% end %>
+    </ul>
```
Comment (nostango) [inline app/views/items/index.html.erb:9]: There is a warning about the file not end in a newline

File: config/routes.rb [hunk 8]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
+  get 'items/new'
+  resources :items
```
Comment (nostango) [inline config/routes.rb:3]: I'm curious to how the routes file with the Devise login will work with this one as we have a main page and a separate database as well.

**Review by niraulap — COMMENTED** [review]
Went through everything! Did not catch anything specific that had not been already caught!

File: test/controllers/items_controller_test.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+require 'test_helper'
+
+class ItemsControllerTest < ActionDispatch::IntegrationTest
+  test "should get new" do
+    get items_new_url
+    assert_response :success
+  end
+
```
Comment (athaichi) [inline test/controllers/items_controller_test.rb:8]: should you check to make sure that you can access items that have already been added to your database?

**Review by athaichi — DISMISSED** [review]

**Review by nostango — APPROVED** [review]

**Pull request merged**